src/site/_includes/components/navigation/index.ts

```diff
@@ -0,0 +1,46 @@
+type MediaQuery = MediaQueryListEvent | MediaQueryList
+
+function navigation() {
+  let menuIsOpen = false
+  const menuButton = document.querySelector<HTMLButtonElement>('.guidance-js-header-toggle')
+  const menuEl = document.querySelector<HTMLElement>('.js-guidance-navigation')
+  if (!(menuButton && menuEl)) {
+    return
+  }
+
+  const mediaQueryList = window.matchMedia('(min-width: 48.0625em)')
+
+  const toggleState = (mediaQuery: MediaQuery) => {
+    if (mediaQuery.matches) {
+      menuEl.removeAttribute('hidden')
+      menuButton.setAttribute('hidden', '')
+    } else {
+      menuButton.removeAttribute('hidden')
+      menuButton.setAttribute('aria-expanded', `${menuIsOpen}`)
+
+      if (menuIsOpen) {
+        menuEl.removeAttribute('hidden')
+      } else {
+        menuEl.setAttribute('hidden', '')
+      }
+    }
+  }
+
+  toggleState(mediaQueryList)
+
+  if ('addEventListener' in mediaQueryList) {
+    mediaQueryList.addEventListener('change', toggleState)
+  } else {
+    // addListener is a deprecated function, however addEventListener
+    // isn't supported by IE or Safari. We therefore add this in as
+    // a fallback for those browsers
+    // @ts-expect-error addEventListener' in mediaQueryList == true in libdom this results in mediaQueryList being never
+    mediaQueryList.addListener(mediaQueryList)
+  }
+  menuButton.addEventListener('click', () => {
+    menuIsOpen = !menuIsOpen
+    toggleState(mediaQueryList)
+  })
+}
+
+export default navigation
```

Comment:
Based on govuk header component, I didn't feel it was necessary to make it a Class and also because it is not a SPA I didn't bother with removing any of the listeners.